### DISCUSSION TITLE
Inheriting R_HOME

### DISCUSSION THREAD

**Discussion by nviets** [mschubert/clustermq#294] [Q&A] [answered]
I run clustermq in an environment with multiple versions of R, and I'd like slurm to spin up workers that naturally inherit the R_HOME of the parent environment. What's the proper way to do this with clustermq? The Q function doesn't seem to support additional arguments, e.g. the following fails:

```
> x <- Q(fx, x=1:10, n_jobs=10, R_HOME = "/nmy/fav/R")
Error in check_args(fun, iter, const) : 
  Argument provided but not accepted by function: R_HOME
```

And the [templates](https://github.com/mschubert/clustermq/blob/master/inst/SLURM.tmpl#L10) rely on the default R of a server.

**Comment by nviets** [comment, marked as answer]
Resolved this by editing the slurm.tmpl file to add ${R_HOME}/bin/R.

**Reply by mschubert** [reply]
Yes, the recommended way is to either set up your `$PATH` to point to the right executable, or to specify it via `$R_HOME` in the template.

The reason for this is that we can't be sure that the same R will be available from the master as will be from the workers, but maybe there's a better solution than the current one.

**Reply by nviets** [reply]
Thanks for the ideas, @mschubert. The slurm template makes it easy to manage through PATH and R_HOME.
